### ISSUE TITLE
App failed to start when graphical environement switch from Unity to Wayland

### ISSUE THREAD

**Issue by klemy** [Foundry376/Mailspring#1067]
##### Are there any related issues?

I think no.

##### What operating system are you using?

- Ubuntu 18.04.1 LTS
- GNOME 3.28.2
- Unity 7.5.0+18.04.20180413-0ubuntu1
- wayland-protocols 1.13-1

##### What version of Mailspring are you using?

1.4.2

--

**Bug?**

##### Is this an email sync or connection issue?

No.

##### Is the issue related to a specific email provider (Gmail, Exchange, etc.)?

No.

##### Is the issue reproducible with a particular attachment, message, signature, etc?

No.

##### So what's the issue?

The app failed to start.

##### How reproduce the issue?

1. Start a session in Unity environment
2. Start Mailspring (it's start well)
3. Close your session and restart it in Wayland environment
4. Start Mailspring (it does not start, logs are provided)
5. Close your session and restart it in Unity environment
6. Start Mailspring (it does not start anymore, logs are provided, exactly the same as step 4)

--

##### Logs (USER has been replaced) :
```
ln: failed to create symbolic link '/home/USER/snap/mailspring/common/snap/mailspring/common/Documents': No such file or directory
ln: failed to create symbolic link '/home/USER/snap/mailspring/common/snap/mailspring/common/Desktop': No such file or directory
ln: failed to create symbolic link '/home/USER/snap/mailspring/common/snap/mailspring/common/Downloads': No such file or directory
ln: failed to create symbolic link '/home/USER/snap/mailspring/common/snap/mailspring/common/Music': No such file or directory
ln: failed to create symbolic link '/home/USER/snap/mailspring/common/snap/mailspring/common/Pictures': No such file or directory
ln: failed to create symbolic link '/home/USER/snap/mailspring/common/snap/mailspring/common/Videos': No such file or directory
ln: failed to create symbolic link '/home/USER/snap/mailspring/common/snap/mailspring/common/Templates': No such file or directory
```

**Comment by duoi** [comment]
Experiencing same thing as @klemy on same OS version, with the added bonus of my entire desktop environment crashing and logging me out when I tried to run Slack.

**Comment by jeroen7s** [comment]
same issue here

**Comment by jeffubelhor** [comment]
I have had this issue as well, trying to diagnose another issue I ran across and switched into Wayland from Gnome (Ubuntu 18.04.1). I also had issues opening up discord, but was able to get Discord working normally by just opening it from the terminal. Before realizing I had switched to Wayland, I assumed it was my doing and made the `.config` directory in `common` manually, which was then automatically populated with files and got rid of the error messages I saw when opening Mailspring from the terminal. However, now running it from the terminal passes back no errors whatsoever and just never finishes opening.

I've installed it through the snap and attempted to remove and reinstall it via Ubuntu Software
